# U.S. Medical Insurance Costs

Goal: explore the data and find relationships between variables.
Answer this questions:
1. What is the average age in the dataset?
2. Whom we have more males or females?
3. Who has the largest and the smallest insurance cost?
4. What is the average insurance cost?
5. What is the average difference between actual and expected insurance cost?

In [17]:
import csv
age, sex, bmi, children, smoker, region, charges = [], [], [], [], [], [], []
with open('insurance.csv') as insurance_csv:
    insurance_data = csv.DictReader(insurance_csv)
    for row in insurance_data:
        age.append(int(row['age']))
        sex.append(row['sex'])
        bmi.append(float(row['bmi']))
        children.append(int(row['children']))
        if row['smoker']=='yes':
            smoker.append(1)
        else:
            smoker.append(0)
        region.append(row['region'])
        charges.append(float(row['charges']))

1. What is the average age in the dataset?

In [26]:
def average_age(age_list):
    total = 0
    for i in age_list:
        total += i
    return round(total/len(age_list), 2)

In [27]:
print(average_age(age))

39.21


2. Whom we have more, males or females?

In [39]:
def males_vs_females(sex_list):
    males = sex_list.count('male')
    females = sex_list.count('female')
    return ('males', males), ('females', females)

In [40]:
print(males_vs_females(sex))

(('males', 676), ('females', 662))


3. Who has the largest and the smallest insurance cost?

In [48]:
def largest_insurance_cost(age_list, sex_list, bmi_list, children_list, smoker_list, region_list, charges_list):
    information = []
    max = -1
    for i in range(len(charges_list)):
        if max < charges_list[i]:
            max = charges_list[i]
            information = [age_list[i], sex_list[i], bmi_list[i], children_list[i], smoker_list[i], region_list[i], charges_list[i]]
    return information

In [49]:
print(largest_insurance_cost(age, sex, bmi, children, smoker, region, charges))

[54, 'female', 47.41, 0, 1, 'southeast', 63770.42801]


In [50]:
def smallest_insurance_cost(age_list, sex_list, bmi_list, children_list, smoker_list, region_list, charges_list):
    information = []
    min = 10000000000000000000000000000000
    for i in range(len(charges_list)):
        if min > charges_list[i]:
            min = charges_list[i]
            information = [age_list[i], sex_list[i], bmi_list[i], children_list[i], smoker_list[i], region_list[i], charges_list[i]]
    return information

In [51]:
print(smallest_insurance_cost(age, sex, bmi, children, smoker, region, charges))

[18, 'male', 23.21, 0, 0, 'southeast', 1121.8739]


4. What is the average insurance cost?

In [54]:
def average_insurance_cost(charges_list):
    total = 0
    for i in charges_list:
        total += i
    return round(total/len(charges_list), 2)

In [55]:
print(average_insurance_cost(charges))

13270.42


5. What is the average difference between actual and expected insurance cost?

In [67]:
def estimated_insurance_cost(age, sex, bmi, children, smoker):
    estimated_cost = 250*age - 128*sex + 370*bmi + 425*children + 24000*smoker - 12500
    return estimated_cost

def average_estimated_insurance_cost(age_list, sex_list, bmi_list, children_list, smoker_list, charges):
    total = 0
    for i in range(len(sex_list)):
        if sex_list[i] == 'male':
            sex_list[i] = 1
        else:
            sex_list[i] = 0
    for i in range(len(age_list)):
        difference = abs(charges[i] - estimated_insurance_cost(age_list[i], sex_list[i], bmi_list[i], children_list[i], smoker_list[i]))
        total += difference
    return round(total/len(age_list), 2)

In [68]:
print(average_estimated_insurance_cost(age, sex, bmi, children, smoker, charges))

4458.18


Answers: 
1. The average age in the dataset is 39 years.
2.  We have almost equal amount of males and females, males - 676, females - 662.
3. The largest insurance cost has 54 year old woman, with a BMI of 47.41, without any children and she smokes, her insurance cost equals 63770.43 dollars. 
The smallest insurance cost has 18 year old boy, with a BMI of 23.21, without any children and  he doesn't smoke, his insurance cost equals 1121.87 dollars.
4. The average insurance cost equals 13270.42 dollars.
5. The average difference between actual and expected insurance cost is 4458.18 dollars.